src/live/src/user-to-user/pubnub-listener.js

```diff
@@ -0,0 +1,56 @@
+import { EventEmitter } from 'events';
+
+/**
+ * @typedef PubNubMessage
+ * @property {string} channel The channel for which the message belongs
+ * @property {string} subscription The channel group or wildcard subscription match (if exists)
+ * @property {string|number} timetoken Publish timetoken
+ * @property {Object} message The Payload
+ */
+
+/**
+ * @typedef PubNubPresence
+ * @property {string} action Can be join, leave, state-change or timeout
+ * @property {string} channel The channel to which the message belongs
+ * @property {string|number} occupancy Number of users connected to the channel
+ * @property {string} state User State
+ * @property {string} subscription The channel group or wildcard subscription match (if exists)
+ * @property {string|number} timestamp Current timetoken
+ * @property {string|number} timetoken Publish timetoken
+ * @property {string[]} uuid UUIDs of users who are connected with the channel
+ */
+
+/**
+ * @typedef PubNubStatus
+ * @property {string} category The status category. An enum is in the PubNub docs. Example: PNConnectedCategory
+ * @property {string} operation The operation the status is for. Example: PNSubscribeOperation
+ * @property {string[]} affectedChannels The channels affected in the operation
+ * @property {string[]} subscribedChannels All the current subscribed channels
+ * @property {string[]} affectedChannelGroups The channel groups affected in the operation
+ * @property {number} lastTimetoken The last timetoken used in the subscribe request
+ * @property {number} currentTimetoken The current timetoken fetched in the subscribe response, which is going to be used in the next request
+ */
+
+export class PubNubListener extends EventEmitter {
+  static statusPrefix = 'status:';
```

Comment:
Yeah I thought about this one too. The point is the user of the class to be able to access it, so they can attach handlers to events in the way they would be emitted.

I'm considering two options here:
1) Add a method, which subscribes to both events by given channel name, which keeps the entire logic for naming the events emitted by `PubNubListener` inside it, which is awesome. But takes away some control. But it is worth exploring.
2) Adding a static getter, without a setter - effectively making the property readonly
3) Do it the way you mentioned, but export the const in order to use it outside

Taking votes here too :)